core/dictgen/src/rootcling_impl.cxx

```diff
@@ -6008,7 +6008,15 @@ int ROOT_rootcling_Driver(int argc, char **argv, const ROOT::Internal::RootCling
    if (nerrors > 0){
       ROOT::TMetaUtils::Info(0,"Problems have been detected during the generation of the dictionary.\n");
    }
+   // Cap the return code to 255, otherwise we magically succeed once we hit 256*N errors as the exit
+   // code of a process is one byte (which overflows at 256).
+   // FIXME: We shouldn't count errors at all via the rootcling return code. See the warning about this from the
+   // libc documentation: http://www.gnu.org/savannah-checkouts/gnu/libc/manual/html_node/Exit-Status.html
+   // This probably requires adapting roottest too as we check the exit code values there.
+   auto mainRetVal = nerrors + retVal;
```

Comment:
I totally agree. I am not aware of roottest checking exit code values, except for "0 or not 0". Can you please set `mainRetVal` to 0 or 1, and see what roottest says about that?

Reply:
I can even point you to an example line where this breaks: https://github.com/root-project/roottest/blob/master/root/meta/genreflex/CMakeLists.txt#L88 :)

But let's indeed remove this error counting mechanic and follow libc recommendations.

Reply:
I stand corrected, but now I have no idea how we passed those tests :)